src/qibocal/protocols/characterization/readout_optimization/twpa_calibration/frequency_SNR.py

```diff
@@ -0,0 +1,231 @@
+from dataclasses import dataclass, field
+from typing import Optional
+
+import numpy as np
+import numpy.typing as npt
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
+from qibolab.platform import Platform
+from qibolab.pulses import PulseSequence
+from qibolab.qubits import QubitId
+from qibolab.sweeper import Parameter, Sweeper, SweeperType
+
+from qibocal.auto.operation import Data, Parameters, Qubits, Results, Routine
+from qibocal.protocols.characterization.utils import HZ_TO_GHZ, V_TO_UV
+
+
+@dataclass
+class ResonatorTWPAFrequencyParameters(Parameters):
+    """ResonatorTWPAFrequency runcard inputs."""
+
+    freq_width: int
+    """Width for frequency sweep relative to the readout frequency (Hz)."""
+    freq_step: int
+    """Frequency step for sweep (Hz)."""
+    min_twpa_freq: int
+    """TPWA frequency minimum value (Hz)."""
+    max_twpa_freq: int
+    """TPWA frequency maximum value (Hz)."""
+    step_twpa_freq: int
+    """TPWA frequency step (Hz)."""
+    nshots: Optional[int] = None
+    """Number of shots."""
+    relaxation_time: Optional[int] = None
+    """Relaxation time (ns)."""
+
+
+@dataclass
+class ResonatorTWPAFrequencyResults(Results):
+    """ResonatorTWPAFrequency outputs."""
+
+    pass
+
+
+ResonatorTWPAFrequencyType = np.dtype(
+    [
+        ("freq", np.float64),
+        ("twpa_freq", np.float64),
+        ("msr", np.float64),
```

Comment:
```suggestion
        ("signal", np.float64),
```
Following #611 there is no `msr` anymore in qibocal. Could you please use `signal` in this PR?

Reply:
Replaced msr by signa